llama_index/llms/openai_like.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Awaitable, Callable, Dict, Optional, Sequence
+
+from llama_index.bridge.pydantic import Field
+from llama_index.callbacks import CallbackManager
+from llama_index.llms import OpenAI
+from llama_index.llms.base import (
+    LLM,
+    ChatMessage,
+    ChatResponse,
+    ChatResponseAsyncGen,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseAsyncGen,
+    CompletionResponseGen,
+    LLMMetadata,
+    llm_chat_callback,
+    llm_completion_callback,
+)
+from llama_index.llms.generic_utils import (
+    achat_to_completion_decorator,
+    acompletion_to_chat_decorator,
+    astream_chat_to_completion_decorator,
+    astream_completion_to_chat_decorator,
+    chat_to_completion_decorator,
+    completion_to_chat_decorator,
+    stream_chat_to_completion_decorator,
+    stream_completion_to_chat_decorator,
+)
+from llama_index.llms.openai_utils import (
+    acompletion_with_retry,
+    completion_with_retry,
+    from_openai_message_dict,
+    to_openai_message_dicts,
+)
+
+
+class OpenAILike(OpenAI):
+    class_type = "openai_like"
+
+    context_window: int = Field(discription="Custom context_window for custom models.")
+    is_chat_model: bool = Field(
+        default=False, description="Indicats that the custom model is a chat_model."
+    )
+    is_function_calling_model: bool = Field(
+        default=False,
+        description="Indicats that the custom model is a function calling model.",
+    )
+
+    def __init__(self, **data: Any) -> None:
+        super().__init__(**data)
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "openai_like_llm"
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        return LLMMetadata(
+            context_window=self.context_window,
+            num_output=self.max_tokens or -1,
+            is_chat_model=self.is_chat_model,
+            is_function_calling_model=self.is_function_calling_model,
+            model_name=self.model,
+        )
+
+    def _use_chat_completions(self, kwargs: Dict[str, Any]) -> bool:
```

Comment:
Instead of overriding this, I think it's better to override `_is_chat_model` to just return `self.is_chat_model`